Rezabekf/default vpc (#81)
aws-samples/cfn101-workshop: workshop/content/20-prerequisites/600-default-vpc.md

[Part 01](/30-workshop-part-01/) of this workshop requires that a default VPC is available in the region you will be deploying CloudFormation templates to.

If you have deleted your default VPC, you can create a new one by following one of the options below.

### 1. Create a default VPC using the Amazon VPC console

1. Open the Amazon VPC console at [https://console.aws.amazon.com/vpc/](https://console.aws.amazon.com/vpc/).
1. In the navigation pane, choose **Your VPCs**.
1. Choose **Actions**, **Create Default VPC**.
1. Choose **Create**. Close the confirmation screen.

### 2. Create a default VPC using the command line

1. Copy the code below to your terminal. Make sure to change the `--region` flag to use a region that you are going to be deploying your CloudFormation to.

       aws ec2 describe-vpcs --filters Name=isDefault,Values=true --query "Vpcs[].VpcId" --region eu-west-2

    If the response is not empty, check that `IsDefault` key is `true` and [move to the next step](/30-workshop-part-01/). If the response shows as `[]` or the VPC is not **default** proceed to the next step.

> 

Reviewer: It would be helpful to assert what you are doing in this step. Why is it necessary?

Why do we want the result `[]`?

Reviewer: We are checking if the default VPC exists. We don't explicitly state that we're doing this. A user less familiar with VPCs may not understand what we're doing.

---
title: 'Default VPC'
date: 2020-02-04T17:34:14Z
weight: 600
---
A default VPC is suitable for getting started quickly, and for launching public instances such as a blog or simple website.

[Part 01](/30-workshop-part-01/) of this workshop requires that a default VPC is available in the region you will be deploying CloudFormation templates to.

> If you have deleted your default VPC, you can create a new one by following one of the options below.

Reviewer: ```suggestion
You will have a default VPC unless you have removed it. If you are unsure, follow the instructions below to check.

If you have deleted your default VPC, you can create a new one by following one of the options below.
```

### 1. Create a default VPC using the Amazon VPC console

1. Open the Amazon VPC console at [https://console.aws.amazon.com/vpc/](https://console.aws.amazon.com/vpc/).
1. In the navigation pane, choose **Your VPCs**.
1. Choose **Actions**, **Create Default VPC**.
1. Choose **Create**. Close the confirmation screen.

### 2. Create a default VPC using the command line

> Let's use the AWS CLI to list all existing VPCs in the region to find out if a default VPC is present or not.

Reviewer: ```suggestion
First we will check if a default VPC is present or not. We will use the AWS CLI to list all existing VPCs in the region
```

1. Copy the code below to your terminal. Make sure to change the `--region` flag to use a region that you are going to be deploying your CloudFormation to.

       aws ec2 describe-vpcs --filters Name=isDefault,Values=true --query "Vpcs[].VpcId" --region eu-west-2

>     If the response is not empty, check that `IsDefault` key is `true` and [move to the next step](/30-workshop-part-01/). If the response is empty `[]` or the VPC is not **default** proceed to the next step.

Reviewer: ```suggestion
If the default VPC exists, it will be included here. Assert that `IsDefault` key is `true` and [move to the next step](/30-workshop-part-01/). You can skip the remainder of this section.

If the response is empty `[]` or the VPC is not **default** proceed to the next step. A default VPC does not exist in this region.
```
